packages/fether-react/src/utils/withHealth.js

```diff
@@ -85,36 +85,34 @@ const rpcs$ = isApiConnected$.pipe(
   take(1),
   switchMap(() =>
     combineLatest(
-      syncStatus$().pipe(
-        map(syncStatus => {
-          if (!syncStatus) {
-            return {
-              isSync: true
-            };
-          }
+      syncStatus$()
+        .pipe(
+          map(syncStatus => {
+            if (!syncStatus) {
+              return {
+                isSync: true
+              };
+            }
 
-          const { currentBlock, highestBlock, startingBlock } = syncStatus;
-          const percentage = currentBlock
-            .minus(startingBlock)
-            .multipliedBy(100)
-            .div(highestBlock.minus(startingBlock));
+            const { currentBlock, highestBlock, startingBlock } = syncStatus;
+            const percentage = currentBlock
+              .minus(startingBlock)
+              .multipliedBy(100)
+              .div(highestBlock.minus(startingBlock));
 
-          return {
-            isSync: false,
-            syncPayload: {
-              currentBlock,
-              highestBlock,
-              percentage,
-              startingBlock
-            }
-          };
-        }),
-        // Emit "not synced" only if we haven't been synced for over 2 seconds,
-        // as syncing to new blocks from the top of the chain usually takes ~1s.
-        // syncStatus$() is distinctUntilChanged, so {isSync: false} will never
-        // be fired twice in a row.
-        switchMap(sync => (sync.isSync ? of(sync) : of(sync).pipe(delay(2000))))
-      ),
+            return {
+              isSync: false,
+              syncPayload: {
+                currentBlock,
+                highestBlock,
+                percentage,
+                startingBlock
+              }
+            };
+          })
+          // Emit "not synced" only if we haven't been synced for over 2 seconds
+        )
+        .pipe(audit(syncStatus => interval(syncStatus.isSync ? 0 : 2000))),
```

Comment:
hmm, I don't get the difference between before and after

Before: on each syncStatus update, do:
- if we're synced, then show that we're synced
- if we're syncing, then wait 2s, then show that we're syncing (if we're still syncing after those 2s).

And new version: on each syncStatus update, do:
- wait 0s if we were synced, then show syncStatus
- wait 2s if we were syncing, then show syncStatus 

Maybe I don't understand audit enough. Does it play the same role as delay here?

ps: would appreciate @axelchalon's eyes on this too

Reply:
Before: on each syncStatus update, do:
- if we're synced, then show that we're synced
- if we're syncing, then wait 2s to emit. The emission is reset if `syncStatus` updates in the mean time (that's the problem).

And new version: on each syncStatus update, do:

- wait 0s if we were synced, then show syncStatus
- wait 2s if we were syncing, then show the most recent syncStatus value.

The main difference is that `audit` will emit anyway (with the most recent value), `delay` will not necessarily emit. Note that `throttle` sits in between and would emit anyway, but with a not up to date value.

BTW I have to thank @axelchalon who help me debug and find this solution :green_heart:

Reply:
Ah nice! got it

Reply:
Yes, the root of the problem was the reasoning

> // syncStatus$() is distinctUntilChanged, so {isSync: false} will never be fired twice in a row.

which was false, because syncStatus$() returns `{isSync: false, currentBlock: ..., highestBlock: ..., startingBlock: ...}` so even though it's distinct until changed it might emit `{isSync: false, ...}` multiple times in a row